libxrdp/xrdp_caps.c

```diff
@@ -577,6 +577,16 @@ xrdp_caps_process_codecs(struct xrdp_rdp *self, struct stream *s, int len)
             g_memcpy(self->client_info.rfx_prop, s->p, i1);
             self->client_info.rfx_prop_len = i1;
         }
+        else if (g_memcmp(codec_guid, XR_CODEC_GUID_IMAGE_REMOTEFX, 16 == 0))
+        {
+            LOG(LOG_LEVEL_INFO, "xrdp_caps_process_codecs: Image RemoteFX(%s), codec id [%d], properties len [%d]",
+                codec_guid_str, codec_id, codec_properties_length);
+            self->client_info.irfx_codec_id = codec_id;
+            i1 = MIN(64, codec_properties_length);
```

Comment:
Do you mean `MAX` rather than `MIN`?

If `codec_properties_length` is 100 (say), the buffer will overflow.

Reply:
Erm, forget the above. It's completely wrong! `MIN` is exactly the right function to use.

Rather than 64, suggest you use `sizeof(self->client_info.irfx_prop)`, as that makes it easier to change the size on the future.

Reply:
Indeed.